### PR TITLE
Fix/262/apply button should be deactivated when you already applied to a project

### PR DESCRIPTION
**Pull request by AxelUZ**
# What does this PR do?
Fixes the error when applying to a project in which the "APPLY" button was not disabled.

# Where should the reviewer start?
app/routes/internshipProjects/$projectId/index.tsx
app/routes/internshipProjects/$projectId/appliedproject.tsx

# How should this be manually tested?
1. Login with linkdin
2. Apply to a project and verify that the button is disabled and the text changes to "APPLIED".

**Once the PR is approved, I will check it in dev which is where the problem is.**

# Any background context you want to provide?

# What are the relevant tickets?
#262 

# Screenshots
Captura de pantalla 2023-11-13 a la(s) 10 40 14 a m

# Questions

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

File: app/routes/internshipProjects/$projectId/index.tsx [hunk 1]
```diff
@@ -77,22 +88,20 @@ export default function ProjectDetail() {
...
+                    width: "200px",
+                    height: "40px",
+                    fontSize: "1em",
+                    margin: 2,
+                  }}
+                  onClick={handleApply} 
+                  disabled={fetcher.state === 'loading' || appliedProjects.includes(projects.name)}
               >
-                APPLY
+                {appliedProjects.includes(projects.name) ? 'APPLIED' : 'APPLY'}
```
Comment (beatriz-sanchez-wizeline) [inline app/routes/internshipProjects/$projectId/index.tsx:104]: This line does not seem to be working, at least for me. It stays as App
Screenshot 2023-11-13 at 13 35 28
ly

Comment (beatriz-sanchez-wizeline) [reply app/routes/internshipProjects/$projectId/index.tsx:104]: Also, I wonder if we could change the style of the button when disabled as it is hard to read the black text on the dark red background. Maybe we could leave it as white but add a blocked cursor instead?

**Comment by beatriz-sanchez-wizeline** [comment]
Hi @AxelUZ, the bug seems to be fixed as the button does not let me apply again if I already applied. That being said, I added a couple of comments on the button label and style in case you are able to do them in the same PR.

File: app/routes/internshipProjects/$projectId/index.tsx (hunk 1 above)
Comment (AxelUZ) [reply app/routes/internshipProjects/$projectId/index.tsx:104]: Thanks @beatriz , I'll check it out 👍

Comment (beatriz-sanchez-wizeline) [reply app/routes/internshipProjects/$projectId/index.tsx]: Working like a charm now :)

**Review by beatriz-sanchez-wizeline — APPROVED** [review]
Sorry for double posting, I am getting used to the GitHub interface for PR review. Everything works fine now :)

File: app/routes/internshipProjects/$projectId/index.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import Header from "../../../core/layouts/Header";
 import type { LoaderArgs } from "@remix-run/server-runtime";
 import { getProjectById } from "~/models/project.server";
-import { Form, useLoaderData } from "@remix-run/react";
+import { Form, useFetcher, useLoaderData } from "@remix-run/react";
```
Comment (martinrobled0) [inline app/routes/internshipProjects/$projectId/index.tsx:4]: I think that you use useFetcher when you dont have a form, for Forms we use useSubmit https://remix.run/docs/en/main/hooks/use-submit

**Pull request merged**